src/object/cli_obj.c

```diff
@@ -143,6 +143,8 @@ obj_free(struct d_hlink *hlink)
 	obj = container_of(hlink, struct dc_object, cob_hlink);
 	D_ASSERT(daos_hhash_link_empty(&obj->cob_hlink));
 	obj_layout_free(obj);
+	if (obj->cob_time_fetch_leader != NULL)
+		D_FREE(obj->cob_time_fetch_leader);
```

Comment:
you can unset cob_time_fetch_leader, so don't need to check it again in obj_layout_create right?

Reply:
The cob_time_fetch_leader is attached to dc_object that will be freed.